src/NuGet.Core/NuGet.PackageManagement/Utility/PackagesConfigLockFileUtility.cs

```diff
@@ -80,6 +83,94 @@ public static string GetPackagesLockFilePath(string projectPath, string nuGetLoc
             return PackagesLockFileUtilities.GetNuGetLockFilePath(projectPath, projectName);
         }
 
+        public static IReadOnlyList<IRestoreLogMessage> ValidatePackagesConfigLockFiles(
+            string projectFile,
+            string packagesConfigFile,
+            string projectName,
+            string nuGetLockFilePath,
+            string restorePackagesWithLockFile,
+            NuGetFramework projectTfm,
+            string packagesFolderPath,
+            bool lockedMode,
+            CancellationToken token)
+        {
+            var lockFilePath = GetPackagesLockFilePath(Path.GetDirectoryName(packagesConfigFile), nuGetLockFilePath, projectName);
+            var lockFileExists = File.Exists(lockFilePath);
+            var lockFileOptIn = MSBuildStringUtility.GetBooleanOrNull(restorePackagesWithLockFile);
+            var useLockFile = lockFileOptIn == true || lockFileExists;
+
+            if (lockFileOptIn == false && lockFileExists)
+            {
+                var message = string.Format(CultureInfo.CurrentCulture, Strings.Error_InvalidLockFileInput, lockFilePath);
+                var errors = new List<IRestoreLogMessage>();
+                var log = RestoreLogMessage.CreateError(NuGetLogCode.NU1005, message, packagesConfigFile);
```

Comment:
I don't understand :(  what properties are you referring to? restorePackagesWithLockFile? they were new properties just for repeatable build and the name seems specific enough to avoid accidentally reusing a property that someone else used for another reason.

Reply:
No, I was referring to WarningsAsErrors. 
Basically would that be respected if the warning was logged in this code path. 

WarningsAsErrors is a property that's shared for build/restore/analyzer warnings.

Reply:
`packages.config` can't be restored from MSBuild, and MSBuild is the only place I can think of that has functionality to elevate or suppress individual codes. So, I think it's out of scope.

Reply:
Yeah, I figured it's not supported, because we would need to elevate the warnings. 

correction, msbuild integrated projects can elevate the nuget errors :)

resolving this.